git_machete/cmd.py

```diff
@@ -16,36 +17,42 @@
 
 # Core utils
 
+T = TypeVar('T')
+
+
 class MacheteException(Exception):
-    def __init__(self, msg, apply_fmt=True):
+    def __init__(self, msg: str, apply_fmt: bool = True) -> None:
         self.parameter = fmt(msg) if apply_fmt else msg
 
-    def __str__(self):
+    def __str__(self) -> str:
         return str(self.parameter)
 
 
-def excluding(iterable, s):
+def excluding(iterable: Iterable[T], s: Iterable[T]) -> List[T]:
     return list(filter(lambda x: x not in s, iterable))
 
 
-def flat_map(func, iterable):
+def flat_map(func: Callable[[T], List[T]], iterable: Iterable[T]) -> List[T]:
     return sum(map(func, iterable), [])
 
 
-def map_truthy_only(func, iterable):
+def map_truthy_only(func: Callable[[T], Optional[T]], iterable: Iterable[T]) -> List[T]:
     return list(filter(None, map(func, iterable)))
 
 
-def non_empty_lines(s):
+def non_empty_lines(s: str) -> List[str]:
     return list(filter(None, s.split("\n")))
 
 
 # Converts a lambda accepting N arguments to a lambda accepting one argument, an N-element tuple.
 # Name matching Scala's `tupled` on `FunctionX`.
-def tupled(f):
+def tupled(f: Callable[..., T]) -> Callable[[Any], T]:
     return lambda tple: f(*tple)
 
 
+get_second: Callable[[Any], T] = tupled(lambda a, b: b)  # type: ignore
```

Comment:
Redefine `get_second` as a regular `def`, and I think it's gonna be type-safe than (possibly another type param `S` will be necessary, tho)

Reply:
Also, use `Tuple[S, T]` or sth like that for type of input param

Reply:
strange... 

```
git_machete/cmd.py:2437: error: Argument 1 to "map" has incompatible type "Callable[[Tuple[T, T]], T]"; expected "Callable[[Tuple[str, str]], T]"
Found 1 error in 1 file (checked 5 source files)
```
for `S` and `T` (the `VarType`s) :confused: 

but since all of the invocations reference strings lets use 
```
def get_second(pair: Tuple[str, str]) -> str:
```
(and since it passes mypy)